### PR TITLE
Tweak start/end/baseline/deviation for SLO log rate analysis.

### PR DESCRIPTION
**Pull request by walterra**
Tweaks the start/end/baseline/deviation for the embedded log rate analysis.

- This aligns the overall time range of the log rate analysis histogram chart with that shown in the `ErrorRatePanel` component. For that to work I had to copy over `getDataTimeRange()` from `x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/error_rate/error_rate_panel.tsx`. This can be consolidated but I didn't know where to best put those two functions in another file.
- I reverted the code to calculate the `intervalFactor` again and to use the `longWindow` for the calculation. Only if the available time range is too short we fall back to `shortWindow`. This way the rest of the code to generate baseline/deviation can stay the same as it is with the treshold alert details page.
- In case of `recovered` alerts, the `kibana.alert.rule.parameters` window cannot be detected correctly. In such a case we now fall back to `LOW_PRIORITY_ACTION_ID=slo.burnRate.low`.

Before:

image

After:

image

Fallback to short window lookback if the available time range is not long enough for long window:

image

### CODE REVIEW

File: x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx [hunk 1]
```diff
@@ -84,14 +105,36 @@ export function LogRateAnalysisPanel({ slo, alert, rule }: Props) {
...
+      ? moment.duration(longWindowValue as number, longWindowUnit as any)
+      : moment.duration(1, 'm');
+  const intervalFactor = Math.max(1, lookbackDuration.asSeconds() / 60);
 
-  const timeRange = {
-    min: alertStart.clone().subtract(15 * intervalFactor, 'minutes'),
-    max: alertEnd ? alertEnd.clone().add(1 * intervalFactor, 'minutes') : moment(new Date()),
-  };
+  const actionGroup = getActionGroupFromReason(alert.reason);
+  const actionGroupWindow = (
```
Comment (mgiota) [inline x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx:131]: @walterra Can't we use here the `alertActionGroup` we define in line 112 above?

Comment (mgiota) [reply x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx:131]: The difference I can see is that `alertActionGroup` above fallbacks to `ALERT_ACTION_ID`, where as the `actionGroup` here fallbacks to whatever the `getActionGroupFromReason` fallbacks, which is `LOW_PRIORITY_ACTION_ID`

Comment (walterra) [reply x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx:131]: The way I understand the alert metadata in line `112` we just get the name like `slo.burnRate.alert`, here we then get the actual metadata of a window including the values for short and long window.

Comment (mgiota) [reply x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx:131]: Sorry I meant to say, can't we use `const relatedWindow = sloWindows.find((window) => window.actionGroup === alertActionGroup);` that I define above? You do pretty much the same for the `actionGroupWindow`

Comment (walterra) [reply x-pack/plugins/observability_solution/slo/public/components/slo/burn_rate/alert_details/components/custom_panels/custom_kql/log_rate_analysis_panel.tsx:131]: Oh now I get what you meant. I tried reusing `relatedWindow`, but then I got different overall time range for the burn rate and log rate analysis chart. I'll leave the code as is for now to keep the time ranges aligned.

**Review by mgiota — APPROVED** [review]

**Pull request merged**